spark/src/main/scala/org/apache/spark/Plugins.scala

```diff
@@ -44,6 +45,20 @@ class CometDriverPlugin extends DriverPlugin with Logging with ShimCometDriverPl
   override def init(sc: SparkContext, pluginContext: PluginContext): ju.Map[String, String] = {
     logInfo("CometDriverPlugin init")
 
+    // register CometSparkSessionExtensions if it isn't already registered
+    val extensionKey = StaticSQLConf.SPARK_SESSION_EXTENSIONS.key
+    val extensionClass = classOf[CometSparkSessionExtensions].getName
+    if (sc.conf.contains(extensionKey)) {
+      val extensions = sc.conf.get(extensionKey)
+      if (!extensions.split(",").map(_.trim).contains(extensionClass)) {
+        sc.conf.set(extensionKey, s"$extensions,$extensionClass")
+      } else {
+        sc.conf.set(extensionKey, extensionClass)
+      }
+    } else {
+      sc.conf.set(extensionKey, extensionClass)
+    }
```

Comment:
its probably would be good to log the overriden value for `spark.sql.extensions` ?